### PR TITLE
Model transformer e2e tests

### PR DESCRIPTION
**Pull request by pavellazar**
#### Description of changes
- added e2e tests for model transformer V2
- bug fixes to assure the e2e tests pass
- implemented on top of #7935 (only the last commit is relevant for this set of changes)

#### Description of how you validated changes
- ran `yarn test`
- ran `SearchableModelTransformerV2`, `IndexTransformer`, and `ModelTransformer` e2e tests
- [build_test_deploy](https://app.circleci.com/pipelines/github/lazpavel/amplify-cli/273/workflows/e5c6d419-09bf-408d-98c8-4adb712b3d9b)

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-model-transformer/src/wrappers/object-definition-wrapper.ts [hunk 1]
```diff
@@ -128,6 +128,17 @@ export class GenericFieldWrapper {
...
   };
+
+  private isListType = (type: TypeNode): boolean => {
+    if (type.kind === Kind.NON_NULL_TYPE) {
+      return isListType(type.type);
+    } else if (type.kind === Kind.LIST_TYPE) {
+      return true;
+    } else {
+      return false;
+    }
```
Comment (cjihrig) [inline packages/amplify-graphql-model-transformer/src/wrappers/object-definition-wrapper.ts:139]: This part can be reduced to `return type.kind === Kind.LIST_TYPE;`

File: packages/graphql-transformers-e2e-tests/src/__tests__/ModelTransformer.e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,812 @@
...
+      title: String!
+      content: String
+      updatedOn: Int # No automatic generation of timestamp if its not AWSDateTime
+    }
+  `;
+  const transformer = new GraphQLTransform({
+    transformers: [new ModelTransformer()],
+  });
+  const out = transformer.transform(validSchema);
+  // fs.writeFileSync('./out.json', JSON.stringify(out, null, 4));
```
Comment (cjihrig) [inline packages/graphql-transformers-e2e-tests/src/__tests__/ModelTransformer.e2e.test.ts:85]: Can you remove this please.

**Review by cjihrig — APPROVED** [review]
LGTM with a couple small nits.

File: packages/graphql-transformers-e2e-tests/src/__tests__/ModelTransformer.e2e.test.ts (hunk 2 above)
Comment (lazpavel) [reply packages/graphql-transformers-e2e-tests/src/__tests__/ModelTransformer.e2e.test.ts]: removed

**Comment by cjihrig** [comment]
@lazpavel this has merge conflicts.

**Pull request merged**